src/app/business/block/bucket-detail/bucket-detail.component.ts

```diff
@@ -528,17 +553,18 @@ export class BucketDetailComponent implements OnInit {
       folderName = this.createFolderForm.value.name + this.colon;
     }
     window['getAkSkList'](()=>{
-      let requestMethod = "PUT";
-      let url = this.BucketService.url + "/" + this.bucketId+ '/' +folderName;
-      window['canonicalString'](requestMethod, url,()=>{
+        let requestMethod = "PUT";
+        let url = "/" + this.bucketId+ '/' +folderName;
+        let requestOptions: any;
         let options: any = {};
-        this.getSignature(options);
+        requestOptions = window['getSignatureKey'](requestMethod, url);
+        options['headers'] = new Headers();
+        options = this.BucketService.getSignatureOptions(requestOptions, options);
```

Comment:
I see these used at multiple places, can't these be part of util function with some customization and re-used

Reply:
These are function calls to the Signature generation method and header preparation. 
`requestOptions = window['getSignatureKey'](requestMethod, url);` returns the signature and signed objects required for the S3 API calls in the `requestOptions` object.

`options = this.BucketService.getSignatureOptions(requestOptions, options)` this is calling the function `getSignatureOptions` from the bucketService that prepares the headers that have been returned by the signature generation method. 
These are the options that can be passed to the signature generation method `getSignatureKey`

`window['getSignatureKey'] = (method, canonicalUri, host?, region?, service?, body?, contentType?, queryString?, headers?)`

Reply:
Yes, so my point was to create a function, which can take one or more params (making some param optional) ..

Reply:
Got it. The above method does exactly that `window['getSignatureKey'] = (method, canonicalUri, host?, region?, service?, body?, contentType?, queryString?, headers?)`

But may be this whole thing can be made into a service. 
I have created an issue #340 to track this.

Reply:
ok